packages/editor/src/components/autocompleters/user.js

```diff
@@ -42,6 +42,11 @@ export default {
 	className: 'editor-autocompleters__user',
 	triggerPrefix: '@',
 
+	allowContext( before ) {
+		// Triggers when at the beginning of a context or when the immediately preceding character is a space.
+		return /^$/.test( before ) || /^\s$/.test( before.slice( -1 ) );
```

Comment:
I wonder if something like a word boundary better represents the use case here? For instance:

```suggestion
		return /\B$/.test( before );
```

This will allow user mentions to be triggered when characters like `:` precede them. I'm not familiar with the autocompleters code to know if it's true though.

Reply:
Thanks for bringing this PR to my attention.

I have tried to see how software and tools that have user mentions with the `@` symbol behave when typing `:@`. Slack, Gmail, and GitHub did not trigger mentions with ":@", so I think it may be more natural to not depend on non-word boundaries.

Reply:
Thanks for testing it out! What about other languages such as Japanese or Chinese? Something like `我和@user` will probably want to match user mention too. But they should also be valid emails, I guess there's no right answer here 😅 .

Reply:
> What about other languages such as Japanese or Chinese?

I used the IME, a common input system for Japanese, to test what happens when the `@` symbol is typed followed by Japanese characters. Interestingly, it had different results.

Slack launches the mention:

https://user-images.githubusercontent.com/54422211/215500741-d1fcfe5c-46d2-4a1d-ab41-4ea715e6c229.mp4

GitHub doesn't launch the mention:

https://user-images.githubusercontent.com/54422211/215500783-26a3c50a-e4ba-4732-be9d-ba1970efd14a.mp4

Gmail doesn't launch the mention (I will not post the video because it contains my personal email address 🙂)

Personally, given that [Gmail allows non-Latin characters as usernames for email addresses](https://blog.google/products/gmail/a-first-step-toward-more-global-email/), I would assume that the `@` symbol following a non-Latin character is an email address and should not trigger a mentions.

Reply:
I guess we'll need second opinions on this. I don't have a strong preference here, but I'm also not the one most familiar with autocompletion either 😅 . We would probably also want to add some documentation if we decide to change its behavior too.

@ellatrix Do you have any insights about this? Or do you have someone in mind who does?

Reply:
No strong opinion. The only thing I have to add is that we should make it very easy to cancel the mention if it's a false positive. Maybe then there's less need for this.

Reply:
This probably only applies to WordPress.com, but it explains that you should put a space before the "@".

https://wordpress.com/support/user-mentions/

There may also be a way to keep the current behavior as is and add UI to toggle the mention feature in the preferences modal 🤔

![image](https://github.com/WordPress/gutenberg/assets/54422211/a078b1ed-23fa-4ab6-b91b-bceb76e70c95)

Reply:
We're not WordPress.com though :) We should make Gutenberg however it makes sense in all languages. I'd lean towards keeping it as is, but it should be super easy to dismiss the mention suggestion. Not sure if that's the case now. Maybe a slight delay until typing stops and a dismiss button?

Alternatively, we could use the word count localisation. If the language counts characters instead of words, don't require space? But I would much rather have it be consistent everywhere.

Reply:
It's very easy to ignore mention suggestions. With the current behaviour, the mention popover will continue to appear as long as there is a partial match to the username, but you can continue typing and there doesn't seem to be any delay. However, some users find the current behavior annoying, as reported in #46348😅

https://github.com/WordPress/gutenberg/assets/54422211/8333764a-6f99-47dd-848d-2aba85c79549

Reply:
Hi all, 

in my opinion, the main problem is that an average user doesn't expect entering @something or email@example.com to autocomplete it from the user table of the website. I have seen people reference external social media accounts/bloggers with @, but never internal.

Why can't this be a feature of the search box that pops up when adding link to text?

Best regards,
Andrej

Reply:
Btw, what is the regex used in PHP that matches these user names? I guess we should match whatever that is doing.

Reply:
Are you referring to the feature where the WordPress username is automatically linked to the profile on the front end?

For example, if it is a forum, it seems that it is converted to a link via a proprietary plugin. It seems that they are expecting a space before the `@`.

https://github.com/WordPress/wordpress.org/blob/46812f1748d395ac7c5b2128a83b2e5d73e1c65c/wordpress.org/public_html/wp-content/plugins/support-forums/inc/class-hooks.php#L826C34-L826C61

I think there is also some sort of process for mentions on the Make WordPress Core blog, but I couldn't find where it was done.

Reply:
Oh, I didn't know we didn't do this in core.

Reply:
I don't have a strong opinion here tbh.